apps/easypid/src/features/proximity/shareDeviceResponse.ts

```diff
@@ -0,0 +1,42 @@
+import { mdocDataTransfer } from '@animo-id/expo-mdoc-data-transfer'
+import { Key, KeyType, TypedArrayEncoder, getJwkFromKey } from '@credo-ts/core'
+import { getMdocContext } from '@credo-ts/core/build/modules/mdoc/MdocContext'
+import { deviceKeyPair } from '@easypid/storage/pidPin'
+import type { EasyPIDAppAgent } from '@package/agent'
+import { DeviceRequest, DeviceResponse, MDoc, type MdocContext, parseIssuerSigned } from '@protokoll/mdoc-client'
+
+type ShareDeviceResponseOptions = {
+  sessionTranscript: Uint8Array
+  deviceRequest: Uint8Array
+  agent: EasyPIDAppAgent
+}
+
+export const shareDeviceResponse = async (options: ShareDeviceResponseOptions) => {
+  const mdocs = await options.agent.mdoc.getAll()
+  const issuerSignedDocuments = mdocs.map((mdoc) => {
+    const docType = mdoc.getTag('DocType') as string
+    return parseIssuerSigned(TypedArrayEncoder.fromBase64(mdoc.base64Url), docType)
+  })
```

Comment:
Can we do a pre-selection on the doctype? So we don't have to fetch all docs? I think the device request should contain some filters?